src/state/reducers/recipesReducer.js

```diff
@@ -0,0 +1,42 @@
+import * as actionTypes from '../actions/actionTypes';
+import { getId } from '../../utils';
+
+const initialState = [];
+
+export const recipesReducer = (state = initialState, action) => {
+  switch(action.type) {
+    case actionTypes.SAVE_RECIPES_SUCCESS: {
+      return state = action.items.map(item => {
+        return {
+          ...item,
+          id: getId(),
```

Comment:
Generally it's not the best approach, since reducers should be pure functions and this line makes it non-pure.

Reply:
But if id is not returned by backend, just leave it as is